Revamped version of Search Widget

yih-en opened the pull request:
### Updates

___Use ```react-measure``` to track the size of screen___
- The previously used ```react-media``` pollutes the pure components and make testing harder

___Revamp mobile design___
- Remove redundant padding and background color
- Make font-size bigger

<img width="318" alt="screen shot 2017-06-13 at 10 42 17 am" src="https://user-images.githubusercontent.com/21019282/27063579-1f0d54e4-5025-11e7-909a-5ae9a8604749.png">

___Autocomplete feature on advanced search___
- Using ```react-select```, autocomplete user input on the advanced search

___Clear all button on the keyword input box___
- Clear all user input when user click the x button

___Real Time Validation in Advanced Search___
- Validate user selection immediately when users do selections

___Updated Demo Page!___
- see https://govwizely.github.io/zAPIness-search-widget/

### Bug Fixes

___Fix Pagination___
- Pagination does not show the last page previously

___Place - (desktop) n/a(mobile) for empty values in result___
- Empty values are confusing in the table view  

___Retain user input when user clicks on the looking glass icon___
- Remember every user input when user types

__

yih-en committed:
Add viewport for the github page (Temporary)

yih-en committed:
Reset offset on every new search

yih-en committed:
Introduce TotalResult component

- Show {start of page} - {end of page} of {total} results in the result page
- Fix missing last page of pagination by replacing _.floor to _.ceil

yih-en committed:
Remove unnecessary files

yih-en committed:
Start works on react-measure

yih-en committed:
Submit keyword when user click on the looking glass

yih-en committed:
ReactMeasure DONE

yih-en committed:
Render mobile result details in non table form

yih-en committed:
Set background color to white

yih-en committed:
Use - instead of n/a in desktop view

yih-en committed:
React-select: autocomplete suggestions for filters

yih-en committed:
Complete dropdown redesign

yih-en committed:
Validation

yih-en committed:
Update README

yih-en committed:
github Page setup

yih-en committed:
x button on keyword search

yih-en committed:
Remove arrow in select input

viclim commented:
<img width="251" alt="screen shot 2017-06-15 at 7 02 10 am" src="https://user-images.githubusercontent.com/949409/27158228-9268eb82-5198-11e7-9636-2e9ddfdc1d5c.png">
@yih-en Make it so that the header doesn't wrap

yih-en committed:
Remove error

yih-en committed:
Only include valid filters during submission

yih-en committed:
Add tests

yih-en committed:
Hide arrow in react-select

- The style will be removed once https://github.com/JedWatson/react-select/pull/1761 is merged to master

yih-en committed:
simple

yih-en committed:
Autocomplete functionality

yih-en committed:
Clear button

yih-en committed:
First Click

yih-en committed:
Clearable Flag

yih-en committed:
Double update

yih-en committed:
Focus input on clear selection

yih-en committed:
Fix Search

yih-en committed:
Use List

yih-en committed:
Disable click handler for no result filter

yih-en committed:
Fix query actions

yih-en committed:
JS lint

yih-en committed:
Fix label
